README.md

```diff
@@ -267,59 +272,150 @@ const UserProfile = ({ userId }) => (
 
 ## React Hooks
 
-### `resource.useEntry`
+### `resource.useState`
+
+A React hook that takes an optional array of arguments and returns a tuple, very much like React's `useState`. Unlike React's `useState`, however, the state is not local to the component. Any other components that are using the state of that same resource get updated immediately with the new state!
+
+```javascript
+const [post, setPost] = postsResource.useState(postId);
+```
 
-A React hook that takes a resource and an optional array of arguments and returns a tuple, very much like React's `useState`. The second item in the tuple works like `useState` in that it sets the in-memory state of the resource. Unlike `useState`, however, the state is not local to the component. Any other components that are using the state of that same resource get updated immediately with the new state!
+Let's look at an example to understand how this works.
 
-Under the hood `react-remote-resource` implements a redux store. Every resource get its own state in the store.
+Take the following resource:
+
+```javascript
+const tweetsResource = createKeyedResource(
+  userId => userId,
+  userId => fetch(`/api/users/${userId}/tweets`).then(res => res.json())
+);
+```
+
+This resource can fetch and store tweets for multiple users. Once it has fetched the tweets for a user it won't re-fetch them. Let's create a component that will render a user's tweets.
 
 ```jsx
-import { createRemoteResouce, useAutoSave } from "react-remote-resource";
-import { savePost, postsResource } from "../resources/posts";
+const UserTweets = ({ userId }) => {
+  const [tweets] = tweetsResource.useState(userId);
 
-const PostForm = ({ postId }) => {
-  const [post, setPost] = postsResource.useEntry(postId);
+  return (
+    <ul>
+      {tweets.map(tweet => (
+        <li key={tweet.id}>
+          <article>
+            <p>{tweet.message}</p>
+            <footer>
+              <small>{tweet.date}</small>
+            </footer>
+          </article>
+        </li>
+      ))}
+    </ul>
+  );
+};
+```
 
-  useAutoSave(post, savePost);
+When `UserTweets` renders, the `useState` hook will take the given `userId`, check if tweets have been fetched for this user, and if they have they will be returned as the first item in the tuple. If they haven't then the loader function will be called and the returned promise will be thrown, which will be caught by the nearest `RemoteResourceBoundary`. Whatever data the promise resolves with will be set as the tweets for that user. Now that the data has been received for this user the component will re-render with the new data. If the component re-renders after this the loader function will not be called since the tweets have already been fetched. We will always get back the latest state.
```

Comment:
```suggestion
When `UserTweets` renders, the `useState` hook will take the given `userId`, and check if tweets have been fetched for this user. If they have, they will be returned as the first item in the tuple. If they haven't then the loader function will be called and the returned promise will be thrown, which will be caught by the nearest `RemoteResourceBoundary`. Whatever data the promise resolves with will be set as the tweets for that user. Once the data has been received for this user, the state will be set and the component will re-render with the new state. If the component re-renders after this the loader function will not be called since the tweets have already been fetched. We will always get back the latest state.
```